baybe/surrogates/base.py

```diff
@@ -83,6 +86,34 @@ def to_botorch(self) -> Model:
 
         return AdapterModel(self)
 
+    @staticmethod
+    def _get_parameter_scaler(
+        parameter: Parameter,
+    ) -> ScalerProtocol | Literal["passthrough"]:  # noqa: F821
+        """Return the scaling method to be used for the given parameter."""
+        return MinMaxScaler()
+
+    def _make_input_scaler(
+        self, searchspace: SearchSpace, measurements: pd.DataFrame
+    ) -> ColumnTransformer:
+        """Make a scaler to be used for transforming computational dataframes."""
+        from sklearn.compose import make_column_transformer
+
+        # Create the composite scaler from the parameter-wise scaler objects
+        # TODO: Filter down to columns that actually remain in the comp rep of the
+        #   searchspace, since the transformer can break down otherwise.
+        transformers = [
+            (self._get_parameter_scaler(p), p.comp_df.columns)
+            for p in searchspace.parameters
+        ]
+        scaler = make_column_transformer(*transformers)
+
+        # TODO: Decide whether scaler is to be fit to parameter bounds and/or
```

Comment:
Couldn't we enable *both*? If one provides measurements, then it should be scaled to the measurements. If one provides the searchspace, scale it according to the searchspace. This would give the user full control and at least to me, it would also be very clear that if I provide measurements, then scaling will be done according to them (potentially with a sanity check regarding the search space). Do you think this could be reasonable?

Reply:
I've deliberately added this TODO to kickstart the discussion here. At its heart, the issue goes back to how scaling and tolerances are to be handled in general, even connected to #114, which is why we put that PR on hold until we get to exactly this point. So I think this needs an alignment in the dev meeting

Reply:
- fit to searchspace bounds (more precicely actually parameter bounds because searchspace might be somewhat reduced due to various other mechanisms) is canonical and will work for small deviations, which are most deviations
- fit to measurements is not canonical, If not note we will not have an issue except for strong outleirs on different scale (which will be rare in the first place). In that case a parameter on linear scale doesnt make sense in the first place

choice for this PR seems pretty obvious to me, im actually surprised that you want to have this discussion here in this PR? In any case the decision can be changed at any point in the future

Reply:
I'm OK with keeping the parameter/searchspace normalization for this PR (i.e. ignoring the measurements, which I'll then drop from the signature again) since this also the way we had it so far, meaning no change involved. And then we can have one final iteration after the workshop and see what to do with #114 and if/how to incorporate that part.

But one thing still a bit unclear: following the same logic, we should actually also take target bounds into consideration whenever they are provided. I can also just keep things as they are for now, meaning that we learn the target standardization from the given measurement data. Just saying that this is actually a bit inconsistent then (and there has been a comment about this in the GP code ever since).

So takeaway for me for this PR: keep parameter normalization based on search space boundaries and target standardization based on actual observed training data. Then, revisit this logic after the workshop, for another release. If that sounds good, please thumbs up 🆙

Reply:
Done